src/components/common/BottomsheetComponent.jsx

```diff
@@ -0,0 +1,314 @@
+  width: 100%;
+  transform: ${({ $level }) => {
+    if ($level === 2) return 'translateY(calc(100% - 51.5%))'; // 50% + 1.5rem
+    if ($level === 3) return 'translateY(calc(100% - 81.5%))'; // 80% + 1.5rem
+    return 'translateY(100%)';
+  }};
+  transition: transform 0.3s cubic-bezier(0.25, 0.46, 0.45, 0.94);
+`;
+
+// 지도 탐색용 바텀시트 컴포넌트
+const MapBottomsheet = () => {
+  const {
+    isMapBottomsheetOpen,
+    mapBottomsheetLevel,
+    mapBottomsheetChildren,
+    setMapBottomsheetLevel,
+    closeMapBottomsheet,
+  } = useBottomsheetStore();
+
+  const startYRef = useRef(0);
+  const startLevelRef = useRef(1);
+  const isDraggingRef = useRef(false);
+
+  const getContentHeight = (level) => {
+    if (level === 1) return '0';
+    if (level === 2) return '40vh';
+    return '70vh';
+  };
+
+  const handlePointerDown = (e) => {
+    e.preventDefault();
+    startYRef.current = e.clientY || e.touches?.[0]?.clientY || 0;
+    startLevelRef.current = mapBottomsheetLevel;
+    isDraggingRef.current = true;
+  };
+
+  const handlePointerMove = (e) => {
+    if (!isDraggingRef.current) return;
+    e.preventDefault();
+  };
+
+  const handlePointerUp = (e) => {
+    if (!isDraggingRef.current) return;
+    isDraggingRef.current = false;
+
+    const currentY = e.clientY || e.changedTouches?.[0]?.clientY || 0;
+    const deltaY = startYRef.current - currentY;
+    const threshold = 50;
+
+    if (Math.abs(deltaY) < threshold) return;
+
+    if (deltaY > 0) {
+      // 위로 스와이프 - 레벨 증가
+      if (startLevelRef.current < 3) {
+        setMapBottomsheetLevel(startLevelRef.current + 1);
+      }
+    } else {
+      // 아래로 스와이프 - 레벨 감소
+      if (startLevelRef.current > 1) {
+        setMapBottomsheetLevel(startLevelRef.current - 1);
+      } else {
+        closeMapBottomsheet();
+      }
+    }
+  };
+
+  // 터치 이벤트 핸들러
+  const handleTouchStart = (e) => {
+    handlePointerDown(e);
+  };
+
+  const handleTouchMove = (e) => {
+    handlePointerMove(e);
+  };
+
+  const handleTouchEnd = (e) => {
+    handlePointerUp(e);
+  };
+
+  if (!isMapBottomsheetOpen) return null;
+
+  return (
+    <SMapBottomsheetLayout>
+      <SMapBottomsheetContainer $level={mapBottomsheetLevel}>
+        <SHandleBar
+          onPointerDown={handlePointerDown}
+          onPointerMove={handlePointerMove}
+          onPointerUp={handlePointerUp}
+          onTouchStart={handleTouchStart}
+          onTouchMove={handleTouchMove}
+          onTouchEnd={handleTouchEnd}
+        />
+        <SContent
+          style={{
+            height: getContentHeight(mapBottomsheetLevel),
+          }}
+        >
+          {mapBottomsheetChildren}
+        </SContent>
+      </SMapBottomsheetContainer>
+    </SMapBottomsheetLayout>
+  );
+};
+
+// 모달 바텀시트 컴포넌트
+const ModalBottomsheet = () => {
+  const {
+    isModalBottomsheetOpen,
+    modalBottomsheetLevel,
+    modalBottomsheetChildren,
+    setModalBottomsheetLevel,
+    closeModalBottomsheet,
+  } = useBottomsheetStore();
```

Comment:
❗ `modalBottomsheetLevel`은 ModalBottomsheet 컴포넌트에서 관리해 주세요. 전역 상태로 선언할 이유가 없어요! 바텀시트 컴포넌트에서만 사용하는 값이에요.
```suggestion
  const {
    isModalBottomsheetOpen,
    modalBottomsheetChildren,
    closeModalBottomsheet,
  } = useBottomsheetStore();
```